feat: improve gemini issue triage workflow

jerop opened the pull request:
This commit improves the gemini issue triage workflow by:

- Passing the available labels as an environment variable to the Gemini CLI.
- Using the output of the Gemini CLI to apply labels to the issue.
- Adding a step to get all repository labels and pass them to the gemini-cli.
- Updating the prompt to classify the issue and output the labels in JSON format.
- Adding a step to apply the labels to the issue using the github-script.

<img width="2258" height="1604" alt="image" src="https://github.com/user-attachments/assets/6e723bdd-cd6c-451f-930b-0310dee9d03a" />

sethvargo commented on .github/workflows/gemini-issue-automated-triage.yml:
```diff
@@ -90,41 +103,92 @@ jobs:
+          github-token: '${{ steps.generate_token.outputs.token || secrets.GITHUB_TOKEN }}'
+          script: |-
+            // Strip code block markers if present
+            let labelsJson = process.env.LABELS_OUTPUT;
```
```suggestion
            const labelsJson = process.env.LABELS_OUTPUT;
```

jerop commented:
@sethvargo addressed all the feedback, thank you!

> is there a reason we don't use `github.rest.issues.setLabels` instead of bifurcating between "add" and "remove"? This only runs when issues are (re)opened, so I'm not sure we need to take on the additional complexity and API calls.

was originally taking a more conservative approach to add new labels and remove some but leave others untouched, but it seems too complex for little benefit, and the user can always implement this if needed -- so switched to the simpler approach using `github.rest.issues.setLabels`
